spatialdata/_core/_spatialdata.py

```diff
@@ -711,6 +823,40 @@ def table(self) -> AnnData:
         """
         return self._table
 
+    @table.setter
+    def table(self, table: AnnData) -> None:
+        """
+        Set the table of a SpatialData object in a object that doesn't contain a table.
+
+        Parameters
+        ----------
+        table
+            The table to set.
+
+        Notes
+        -----
+        If a table is already present, it needs to be removed first.
+        The table needs to pass validation (see :class:`~spatialdata.TableModel`).
+        If the SpatialData object is backed by a Zarr storage, the table will be written to the Zarr storage.
+        """
+        TableModel().validate(table)
+        if self.table is not None:
+            raise ValueError("The table already exists. Use del sdata.table to remove it first.")
+        self._table = table
+        if self.is_backed():
+            store = parse_url(self.path, mode="r+").store
+            root = zarr.group(store=store)
+            write_table(table=self.table, group=root, name="table")
+
+    @table.deleter
+    def table(self) -> None:
+        """Delete the table."""
+        self._table = None
+        if self.is_backed():
+            store = parse_url(self.path, mode="r+").store
+            root = zarr.group(store=store)
+            del root["table"]
```

Comment:
could you elaborate on the use case of this?

Reply:
In the setter I don't allow to overwrite a table by default to prevent accidental data loss. I do the same in `add_image()`, `add_labels()`, etc. In those functions I could add a `overwrite` parameter, to be sure that the user wanted to replace the element, but with a setter I could not. So if the user tries to replace an existing table I raise an exception in which I ask to explicitly delete the table first.